### PR TITLE
[Server] Implement Mastodon API integration and set up the Gun.js database

### PR DESCRIPTION
**Pull request by keran-w**
# Overview
The server is designed to fetch data from the external Mastodon API and save the necessary data in the Gun.js database. Currently, the API includes getting specific account information, account searches/finds, follower/following lists, post lists, and post details. [OpenAPI 3 standard Documentation](http://localhost:10000/api/v1/docs/) provides instructions for `GET /api/v1/account/{id}/initialize`, and more documentation will be provided as needed.

# Future Work
- Create structured data models for accounts, posts, etc. and integrate with the Gun.js database
- Develop controllers for data management -- save, access and update
- Detailed documentation of each API route -- error handling, expected response, parameters. Make sure easy to understand

### CODE REVIEW

File: API Response Example/initialize_109252111498807689.json
Comment (alexluo1998) [inline API Response Example/initialize_109252111498807689.json:1]: Let's get rid of this since it represents out-dated structure

File: .gitignore
Comment (alexluo1998) [inline .gitignore:33]: would this ignore package.json...?

File: API Response Example/initialize_109252111498807689_adjacency.json
Comment (alexluo1998) [inline API Response Example/initialize_109252111498807689_adjacency.json:1]: Tbh not sure if this is needed since we have Swagger. But we can keep it for now in case later we want to include it in admin folder as part of the design doc

File: server/tsconfig.json
Comment (alexluo1998) [inline server/tsconfig.json:1]: I removed this in my PR (as well as the TypeScript dependency) since I am no sure how it's used and seems like it's not working in my IDE. If you want to add it back, move this config into package.json (you may need to google the proper key, for example, for prettier config it's "prettier") so that we don't have messy config files all in the root folder

File: server/src/utils/http.js
Comment (alexluo1998) [inline server/src/utils/http.js:3]: One problem I noticed during my refactoring is that the server name is hardcoded to be mastodon.social. I guess it could work because mastodon.social is the most popular server and ususally has data linked with other servers, but I am not sure if there is any edge case.
Check my PR to see how I made the server name to be part of the Mastodon API (Adaptor)

**Review by alexluo1998 — CHANGES_REQUESTED** [review]
1. the db and mastodon folder will be redundant after my PR is merged. The remaining work, is to link the DB with the mastodon api calls - I did it but for the old json contract (and I have it in my local - didn't include in my PR because they represent out-dated work). We can get on a call to sync up and I will help you sort out the things that should be kept. (also take a look at my recording)
2. Unit test, unfortunately, is still required. I have tons of examples in my PR so I hope it will help provide some ideas on what do write.
3. run prettier and linter once my PR is merged
4. JSDocs (can use GPT to generate as well)

Let's get this cleaned up and unblock it from merging
Thank you so much for all the good work so far

File: .gitignore
Comment (keran-w) [reply .gitignore:33]: Fixed. Igone only *.json from Examples folder

File: server/tsconfig.json
Comment (keran-w) [reply server/tsconfig.json:1]: This is for static code analysis only. We can remove this file.

File: server/src/services/accountService.js
Comment (keran-w) [reply server/src/services/accountService.js:86]: Removed

**Review by Untellable — CHANGES_REQUESTED** [review]

File: server/src/services/accountService.js
Comment (alexluo1998) [reply server/src/services/accountService.js:97]: This is actually a shorthand of
```
    const { data } = await api.getAccountFollowing(id);
    const data = response.data;
```
It is not a good way to unwrap object because
1. the variable name can only be 'data' because that's actually the JSON property name (if you change it to a different name, it will cause an error) - and as Ben mentioned, 'data' is not a very clear name
2. if the response is null (for example when an error was thrown), this will cause an error because you are accessing a property of undefined object

The proper way to do it is something like
```
    const response = await api.getAccountFollowing(id);
    if (response && response.data) {
       const followingUserInfo = response.data;
    }
```

**Review by alexluo1998 — CHANGES_REQUESTED** [review]
This branch currently doesn't have the latest change in main. Let's create a clean branch from main, and then move your changes to it

To help you summarize the changes you need to move, and it's actually not much:
1. ./api/router.js -> add your /initialize endpoint to it
2. ./api/initialize/initialize-handler.js -> move the error handling code in your router.js into a separate file (see my code in main)
3. ./service/account-service.js
One thing I forgot to mention is that we will use all lower case letters plus dash (-) for js file names. This is what I found to be the most common style for js projects.

**Pull request closed without merging**
